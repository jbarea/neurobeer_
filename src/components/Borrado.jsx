import React, { Component } from 'react';

class Borrado extends Component {
    constructor(props){
        super(props);
        var datosBorrado = JSON.parse(localStorage.getItem("datos"));
        var id = this.props.match.params.id;
        var ele = JSON.parse(localStorage.getItem("datos")).findIndex(function (iteracion) {
            return iteracion.id === id;
        });
        datosBorrado.splice(ele,1);
        //console.log(ele);
        localStorage.removeItem("datos");
        localStorage.setItem("datos", JSON.stringify(datosBorrado));
        alert('Se ha eliminado la entrada correctamente!!!');
    } 

    render(){
        return(
            <div></div>
        );
    }
}

export default Borrado;