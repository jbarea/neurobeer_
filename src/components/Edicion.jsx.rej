diff a/src/components/Edicion.jsx b/src/components/Edicion.jsx	(rejected hunks)
@@ -14,7 +14,8 @@
         });
         console.log(data);
         console.log(id_);
-        console.log('props from link', this.props.match.params.id)
+        console.log('props from link', this.props.match.params.id);
+        console.log("valor de data.imagen: "+data.imagen);
         this.state = {
             elem: ele,
             id: data.id,
