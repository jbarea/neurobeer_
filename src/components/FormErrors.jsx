import React from 'react';
import '../css/form-errors.css';

export const FormErrors = ({formErrors}) => {
    return(
    <div className="form-errors">
        {Object.keys(formErrors).map((fieldName, i) => {
            //console.log("valor del indice de iteracion:"+i);
            //console.log("contenido de fieldName:"+fieldName);
            if(formErrors[fieldName].length > 0){
                return (
                    <p key={i}>{fieldName} {formErrors[fieldName]}</p>
                )
            } else {
                return '';
            }
        })}
    </div>
    )
}