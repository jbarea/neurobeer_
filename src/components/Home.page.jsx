import React, { Component } from 'react';
import '../css/estilos.css';
//import imagen from '../img/cerveza.jpg';
class Home extends Component {
    render() {
        return(
            <div className="cuerpo">
                {/* <img src={imagen} width='50%' height='100%' alt='imagen de un barril de cerveza para el fondo de la pagina'></img> */}
                <div className="imagen-fondo">
                
                </div>
            </div> 
        )
    }
}

export default Home;