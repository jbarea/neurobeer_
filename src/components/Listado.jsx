import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../css/estilos.css';
import '../css/listado.css';
import { ordenarPor } from '../utils/actions';
import { seleccionTipo } from '../data/selectOptions';
import uuid from 'uuid';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPencilAlt, faEraser, faArrowUp, faArrowDown } from '@fortawesome/free-solid-svg-icons'; //faArrowUp y faArrowDown para flechas de ordenacion

class Listado extends Component {
    constructor(props) {
        super(props);
        this.state = {
            porPrecio: false,
            porTipo: false,
            porNombre: false,
            datosListado: JSON.parse(localStorage.getItem("datos")) || []
        }
    }

    handleOrder = (opcion) => {
        switch(opcion){
            case 'graduacion':
                this.setState(this.state.datosListado.sort(ordenarPor("graduacion",true,parseFloat)));
                //console.log(this.state.datosListado);
                break;
            case 'graduacionD':
                this.setState(this.state.datosListado.sort(ordenarPor("graduacion",false,parseFloat)));
                break;
            case 'tipo':
                this.setState(this.state.datosListado.sort(ordenarPor("tipo",true)));
                break;
            case 'tipoD':
                this.setState(this.state.datosListado.sort(ordenarPor("tipo",false)));
                break;
            case 'nombre':
                this.setState(this.state.datosListado.sort(ordenarPor("nombre",true)));
                break;
            case 'nombreD':
                this.setState(this.state.datosListado.sort(ordenarPor("nombre",false)));
                break;
            case 'id':
                this.setState(this.state.datosListado.sort(ordenarPor("id",true)));
                break;
            case 'idD':
                this.setState(this.state.datosListado.sort(ordenarPor("id",false)));
                break;
            case 'precio':
                this.setState(this.state.datosListado.sort(ordenarPor("precio",true,parseFloat)));
                break;
            case 'precioD':
                this.setState(this.state.datosListado.sort(ordenarPor("precio",false,parseFloat)));
                //console.log(this.state.datosListado);
                break;
            default:
                break;
        }
    }

    render() {
        return(
        <div className="listadoContainer">
            <div>
                <div className="tablaListado titulosTabla" key={uuid()}>
                    <div className="elementoLista">ID
                        <FontAwesomeIcon icon={faArrowUp} onClick={() => this.handleOrder("id")}/>
                        <FontAwesomeIcon icon={faArrowDown} onClick={() => this.handleOrder("idD")}/>
                    </div>
                    <div className="elementoLista">Imagen</div>
                    <div className="elementoLista">Nombre
                        <FontAwesomeIcon icon={faArrowUp} onClick={() => this.handleOrder("nombre")} />
                        <FontAwesomeIcon icon={faArrowDown} onClick={() => this.handleOrder("nombreD")} />
                    </div>
                        <div className="elementoLista" >Grad
                        <FontAwesomeIcon icon={faArrowUp} onClick={() => this.handleOrder("graduacion")}/>
                        <FontAwesomeIcon icon={faArrowDown} onClick={() => this.handleOrder("graduacionD")}/>
                    </div>
                    <div className="elementoLista">Tipo
                        <FontAwesomeIcon icon={faArrowUp} onClick={() => this.handleOrder("tipo")} />
                        <FontAwesomeIcon icon={faArrowDown} onClick={() => this.handleOrder("tipoD")} />
                    </div>
                    <div className="elementoLista">Ingredientes</div>
                    <div className="elementoLista">Sin Gluten</div>
                    <div className="elementoLista">Precio
                        <FontAwesomeIcon icon={faArrowUp} onClick={() => this.handleOrder("precio")} />
                        <FontAwesomeIcon icon={faArrowDown} onClick={() => this.handleOrder("precioD")} />
                    </div>
                    {
                            /* this.porPrecio ? ordenarPorPrecio(this.state.datosListado, "precio") : this.porTipo ? ordenarPorTipo(this.state.datosListado, "tipo") :
                                this.porNombre ? ordenarPorNombre(this.state.datosListado, "nombre") : this.state.datosListado */
                    }
                </div>
                {
                    this.state.datosListado.map((item) => 
                    <div className="tablaListado" key={uuid()}>
                        <div className="elementoLista" key={item.id}>
                            <Link className="botonEditar" to={`/edicion/${item.id}`} state={item.id}><FontAwesomeIcon icon={faPencilAlt} /></Link>
                            <Link className="botonBorrar" to={`/borrado/${item.id}`} state={item.id}><FontAwesomeIcon icon={faEraser} /></Link>
                            {/* <FontAwesomeIcon icon={faPencilAlt}/><Link className="botonEditar" to={`/edicion/${item.id}`} state={item.id}>Editar</Link> */}
                            {/* <FontAwesomeIcon icon={faEraser}/><Link className="botonBorrar" to={`/borrado/${item.id}`} state={item.id}>Borrar</Link> */}
                        {item.id}</div>
                        <div className="elementoLista" >{item.imagen}</div>
                        <div className="elementoLista" >{item.nombre}</div>
                        <div className="elementoLista" >{item.graduacion}</div>
                        <div className="elementoLista" >{seleccionTipo[item.tipo]}</div>
                        <div className="elementoLista" >{item.ingredientes}</div>
                        <div className="elementoLista" >{item.check === true ? "Sin Gluten" : ""}</div>
                        <div className="elementoLista" >{item.precio}</div>
                    </div>
                    )
                }
            </div>
        </div>
        )
    }
 }

export default Listado;