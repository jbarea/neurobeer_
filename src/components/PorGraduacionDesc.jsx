import React, { Component } from 'react';
import uuid from 'uuid';
import { Link } from 'react-router-dom';
import '../css/listado.css';
import '../css/estilos.css';
import { ordenarPor } from '../utils/actions';
import { seleccionTipo } from '../data/selectOptions';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPencilAlt, faEraser, faArrowUp, faArrowDown } from '@fortawesome/free-solid-svg-icons';

class PorGraduacion extends Component {
    constructor(props) {
        super(props);
        this.state = {
            datosListado: JSON.parse(localStorage.getItem("datos")) || []
        }
        this.state.datosListado.sort(ordenarPor("graduacion", false, parseFloat));
    }

    render() {
        return (
            <div className="listadoContainer">
                <div>
                    <div className="tablaListado titulosTabla" key={uuid()}>
                        <div className="elementoLista"><Link to={`/listado/porID/`}>ID</Link></div>
                        <div className="elementoLista">Imagen</div>
                        <div className="elementoLista"><Link to={`/listado/porNombre`}>Nombre</Link></div>
                        <div className="elementoLista">Grad
                            <Link to={`/listado/porGraduacion/`}>
                                <FontAwesomeIcon icon={faArrowUp} />
                            </Link>
                            <Link to={`/listado/porGraduacionDesc/`}>
                                <FontAwesomeIcon icon={faArrowDown} />
                            </Link>
                        </div>
                        <div className="elementoLista"><Link to={`/listado/porTipo/`}>Tipo</Link></div>
                        <div className="elementoLista">Ingredientes</div>
                        <div className="elementoLista">Sin Gluten</div>
                        <div className="elementoLista"><Link to={`/listado/porPrecio`}>Precio</Link></div>
                    </div>
                    {
                        this.state.datosListado.map((item) =>
                            <div className="tablaListado" key={uuid()}>
                                <div className="elementoLista" key={item.id}>
                                    <Link className="botonEditar" to={`/edicion/${item.id}`} state={item.id}><FontAwesomeIcon icon={faPencilAlt} /></Link>
                                    <Link className="botonBorrar" to={`/borrado/${item.id}`} state={item.id}><FontAwesomeIcon icon={faEraser} /></Link>
                                    {item.id}</div>
                                <div className="elementoLista" >{item.imagen}</div>
                                <div className="elementoLista" >{item.nombre}</div>
                                <div className="elementoLista" >{item.graduacion}</div>
                                <div className="elementoLista" >{seleccionTipo[item.tipo]}</div>
                                <div className="elementoLista" >{item.ingredientes}</div>
                                <div className="elementoLista" >{item.check === true ? "Sin Gluten" : ""}</div>
                                <div className="elementoLista" >{item.precio}</div>
                            </div>
                        )
                    }
                </div>
            </div>
        )
    }
}

export default PorGraduacion;