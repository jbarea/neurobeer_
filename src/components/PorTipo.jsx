import React, { Component } from 'react';
import uuid from 'uuid';
import { Link } from 'react-router-dom';
import '../css/estilos.css';
import '../css/listado.css';
import { ordenarPor } from '../utils/actions';
import { seleccionTipo } from '../data/selectOptions';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPencilAlt, faEraser } from '@fortawesome/free-solid-svg-icons';

class PorTipo extends Component {
    constructor(props){
        super(props);
        this.state = {
            datosListado: JSON.parse(localStorage.getItem("datos")) || []
        }
        console.log(this.state.datosListado);
        this.state.datosListado.sort(ordenarPor("tipo", true));
        console.log(this.state.datosListado)

    }

    render(){
        return(
            <div className="listadoContainer">
                <div>
                    <div className="tablaListado titulosTabla" key={uuid()}>
                        <div className="elementoLista">ID</div>
                        <div className="elementoLista">Imagen</div>
                        <div className="elementoLista"><Link to={`/listado/porNombre/`}>Nombre</Link></div>
                        <div className="elementoLista"><Link to={`/listado/porGraduacion/`}>Grad</Link></div>
                        <div className="elementoLista"><Link to={`/listado/porTipo/`}>Tipo</Link></div>
                        <div className="elementoLista">Ingredientes</div>
                        <div className="elementoLista">Sin Gluten</div>
                        <div className="elementoLista">Precio</div>
                    </div>
                    {
                        this.state.datosListado.map((item) =>
                            <div className="tablaListado" key={uuid()}>
                                <div className="elementoLista" key={item.id}>
                                    <FontAwesomeIcon icon={faPencilAlt} /><Link to={`/edicion/${item.id}`} state={item.id}>Editar</Link>
                                    <FontAwesomeIcon icon={faEraser} /><Link to={`/borrado/${item.id}`} state={item.id}>Borrar</Link>
                                    {item.id}</div>
                                <div className="elementoLista" >{item.imagen}</div>
                                <div className="elementoLista" >{item.nombre}</div>
                                <div className="elementoLista" >{item.graduacion}</div>
                                <div className="elementoLista" >{seleccionTipo[item.tipo]}</div>
                                <div className="elementoLista" >{item.ingredientes}</div>
                                <div className="elementoLista" >{item.check === true ? "Sin Gluten" : ""}</div>
                                <div className="elementoLista" >{item.precio}</div>
                            </div>
                        )
                    }
                </div>
            </div>
        )
    }
}

export default PorTipo;