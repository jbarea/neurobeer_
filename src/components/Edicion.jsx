import React, { Component } from 'react';
import { FormErrors } from './FormErrors';
import '../css/edicion.css';

class Edicion extends Component {
    constructor(props){
        super(props);
        var id_ = this.props.match.params.id;
        var data = JSON.parse(localStorage.getItem("datos")).find(function (iteracion) {
            return iteracion.id === id_;
        });
        var ele = JSON.parse(localStorage.getItem("datos")).findIndex(function (iteracion) {
            return iteracion.id === id_;
        });
        console.log(data);
        console.log(id_);
        console.log('props from link', this.props.match.params.id)
        this.state = {
            elem: ele,
            id: data.id,
            imagen: '',
            grad: data.graduacion,
            nombre: data.nombre,
            tipo: data.tipo,
            ingr: data.ingredientes,
            chk: data.check,
            precio: data.precio,
            formErrors: { imagen: '', grad: '', nombre: '', tipo: '', ingr: '', precio: '' },
            imagenValid: false,
            gradValid: false,
            nombreValid: false,
            tipoValid: false,
            ingrValid: false,
            chkValid: false,
            precioValid: false,
            formValid: true,
            beerStore: JSON.parse(localStorage.getItem("datos")) || []
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        let data_ = {
            id: this.state.id,
            imagen: this.state.imagen,
            graduacion: this.state.grad,
            nombre: this.state.nombre,
            tipo: this.state.tipo,
            ingredientes: this.state.ingr,
            check: this.state.chk,
            precio: this.state.precio
        }
        //this.state.beerStore.push(data_);
        var ele = this.state.elem;
        console.log(ele);
        this.state.beerStore[ele] = data_;
        localStorage.setItem('datos', JSON.stringify(this.state.beerStore));
        this.setState(prevState => {
            return prevState = {
                ...prevState,
                imagen: '',
                grad: '',
                nombre: '',
                tipo: '',
                ingr: '',
                chk: '',
                precio: '',
                formValid: false
            }
        })
        alert("Se ha modificado la entrada correctamente!. Pulse aceptar para continuar.");
    }

    readURL(input) {
        document.getElementById("imgSelected").style.display = "block";

        if (input.files && input.files[0]) {
            var reader = new FileReader();

            reader.onload = function (e) {
                document.getElementById('imgSelected').src = e.target.result;
            }

            reader.readAsDataURL(input.files[0]);
        }
    }

    handleUserInput = (e) => {
        e.preventDefault();
        const name = e.target.name;
        const value = e.target.type === 'checkbox' ? e.target.checked : e.target.value;
        
        this.setState({ [name]: value },
            () => { this.validateField(name, value) })
    }

    validateField(fieldName, value) {
        let fieldValidationErrors = this.state.formErrors;
        let {
            imagenValid, gradValid, nombreValid, tipoValid, ingrValid, chkValid, precioValid
        } = this.state;

        switch (fieldName) {
            case 'imagen':
                this.imagen = value;
                imagenValid = value.match(/[0-9a-zA-Z]/i);
                fieldValidationErrors.imagen = imagenValid ? '' : 'Error al seleccionar archivo!';
                break;
            case 'grad':
                //this.grad = value;
                gradValid = value.match(/[0-9]/i);
                fieldValidationErrors.grad = gradValid ? '' : 'El campo Graduación debe ser numérico!';
                break;
            case 'nombre':
                //this.nombre = value;
                nombreValid = value.match(/[a-zA-Z]/i);
                fieldValidationErrors.nombre = nombreValid ? '' : 'El campo solo puede contener caracteres alfanuméricos';
                break;
            case 'tipo':
                //seleValid = value.match(/[0-9]/i);
                tipoValid = document.getElementById("tipo").selectedIndex;
                let tipo2 = document.getElementById("tipo");
                this.tipo = tipo2.options[tipo2.selectedIndex].text;
                console.log(this.tipo);
                if ((tipoValid === null) || (tipoValid === 0)) {
                    tipoValid = false;
                } else {
                    tipoValid = true;
                }
                console.log(tipoValid);
                fieldValidationErrors.tipo = tipoValid ? '' : 'Debe de seleccionar un tipo';
                break;
            case 'ingr':
                ingrValid = value.match(/[a-zA-Z]/i);
                fieldValidationErrors.ingr = ingrValid ? '' : 'El campo solo puede contener caracteres alfanuméricos';
                break;
            case 'chk':
                chkValid = value;
                console.log(chkValid);
                /* if(!chkValid.checked){
                    chkValid = false;
                }else{
                    chkValid = true;
                } */
                break;
            case 'precio':
                precioValid = value.match(/[0-9]/i);
                fieldValidationErrors.precio = precioValid ? '' : 'El campo precio debe de ser numérico';
                break;
            default:
                break;
        }

        this.setState(
            {
                formErrors: fieldValidationErrors,
                imagenValid: imagenValid,
                gradValid: gradValid,
                nombreValid: nombreValid,
                tipoValid: tipoValid,
                ingrValid: ingrValid,
                chkValid: chkValid,
                precioValid: precioValid
            }, this.validateForm);
    }

    validateForm() {
        this.setState({
            formValid: this.state.imagenValid && this.state.gradValid && this.state.nombreValid && this.state.tipoValid &&
                this.state.ingrValid && this.state.precioValid
        });
    }

    errorClass(error) {
        return (error.length === 0 ? '' : 'has-error');
    }

    render(){
        
        return(
            <form className="formBeerEdit" method="POST" enc="multipart/form-data" onSubmit={this.handleSubmit}>
                <h2>Edite los valores necesarios:</h2>
                <div className="formBorder">
                    <div className="panel panel-default">
                        <FormErrors formErrors={this.state.formErrors}/>
                    </div>
                    <div className="form-group" id="img">
                        <label htmlFor="imagen">Imagen:</label>
                        <input type="file" className="form-control-file" id="img_in" name="imagen" value={this.state.imagen}
                            onChange={(event) => this.handleUserInput(event)}></input>
                        <img id="imgSelected" src="#" name="imgSelected" alt="imagen seleccionada mediante cuadro de diálogo"></img>
                        <br />
                    </div>
                    <div className="form-row">
                        <div className={`form-group col-md-4 ${this.errorClass(this.state.formErrors.nombre)}`} id="nom">
                            <label htmlFor="nombre">Nombre:</label>
                            <input type="text" className="form-control" name="nombre" value={this.state.nombre}
                                onChange={(event) => this.handleUserInput(event)} required></input><br />
                        </div>
                        <div className={`form-group col-md-2 ${this.errorClass(this.state.formErrors.grad)}`} id="grad">
                            <label htmlFor="grad">Graduación:</label>
                            <input type="text" className="form-control" size="3" name="grad" value={this.state.grad}
                                onChange={(event) => this.handleUserInput(event)} required></input><br />
                        </div>
                    </div>
                    <div className="form-group" id="sel">
                        <label htmlFor="sele">Selección</label>
                        <select id="tipo" name="tipo" onChange={(event) => this.handleUserInput(event)}>
                            <option value="0">Seleccione Un tipo</option>
                            <option value="1">De trigo</option>
                            <option value="2">Porter y stout</option>
                            <option value="3">Ale</option>
                            <option value="4">Lager</option>
                            <option value="5">Lambic</option>
                        </select><br />
                    </div>
                    <div className={`form-group ${this.errorClass(this.state.formErrors.ingr)}`} id="ingr">
                        <label htmlFor="ingr">Ingredientes:</label><br />
                        <textarea name="ingr" rows="4" cols="80" value={this.state.ingr}
                            onChange={(event) => this.handleUserInput(event)}></textarea><br />
                    </div>
                    <div className="form-group" id="check">
                        <label><input type="checkbox" name="chk" id="chk" value={this.state.chk}
                            onChange={(event) => this.handleUserInput(event)}></input>Sin gluten</label><br />
                    </div>
                    <div className={`form-group ${this.errorClass(this.state.formErrors.precio)}`} id="pre">
                        <label htmlFor="precio">Precio:</label><br />
                        <input type="text" name="precio" value={this.state.precio}
                            onChange={(event) => this.handleUserInput(event)} required></input><br />

                    </div>
                    <div className="form-group">
                        <button type="submit" className="btn btn-primary" align="center">
                            Aceptar
                        </button>
                    </div>
                </div>
            </form>
        )
    }
}

export default Edicion;