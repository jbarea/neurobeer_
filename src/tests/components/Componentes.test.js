import React from 'react';
//import { shallow } from 'enzyme';
import ReactTestUtils from 'react-dom/test-utils';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Home from '../../components/Home.page';
import Form from '../../components/Form';
import Listado from '../../components/Listado';
import Edicion from '../../components/Edicion';
import Borrado from '../../components/Borrado';

import { FormErrors } from '../../components/FormErrors';


describe('components > Form.jsx', () => {
    test('Deberia renderizar el formulario con ReactShallowRenderer', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<Form />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    }),

    test('Comprobamos que se renderiza el formulario y todos los campos', () => {

    })
});

describe('components > Home.page.jsx', () => {
    test('Debería renderizar el formulario con ReactShallowRenderer', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<Home />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    })
})

describe('components > FormErrors.jsx', ()=> {
    test('Debería renderizar el formulario con ReactShallowRenderer', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<FormErrors />);
        const result = renderer.getRenderOutput();
        expect(result.type).toBe('div');
    })
})

describe('components > Listado.jsx', () => {
    test('Debería renderizar el formulario con ReactShallowRenderer', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<Listado />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    })
})

describe('components > Edicion.jsx', () => {
    test('Debería renderizar el formulario con ReactShallowRenderer', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<Edicion />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    })
})

describe('components > Borrado.jsx', () => {
    test('Debería renderizar el formulario con ReactShallowRenderer', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<Borrado />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    })
})
