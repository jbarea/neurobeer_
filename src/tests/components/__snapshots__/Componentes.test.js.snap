// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Form.jsx Comprobamos que se renderiza bien nuestro componente Form 1`] = `
<form
  className="formBeer"
  encType="multipart/form-data"
  method="POST"
  onSubmit={[Function]}
>
  <h2>
    Introduzca valores para la creación:
  </h2>
  <div
    className="formBorder"
  >
    <div
      className="panel panel-default"
    >
      <FormErrors
        formErrors={
          Object {
            "grad": "",
            "imagen": "",
            "ingr": "",
            "nombre": "",
            "precio": "",
            "tipo": "",
          }
        }
      />
    </div>
    <div
      className="form-group"
      id="img"
    >
      <label
        htmlFor="imagen"
      >
        Imagen:
      </label>
      <input
        className="form-control-file"
        id="img_in"
        name="imagen"
        onChange={[Function]}
        required={true}
        type="file"
        value=""
      />
      <img
        alt="imagen seleccionada mediante cuadro de diálogo"
        id="imgSelected"
        name="imgSelected"
        src="#"
      />
      <br />
    </div>
    <div
      className="form-row"
    >
      <div
        className="form-group col-md-4 "
        id="nom"
      >
        <label
          htmlFor="nombre"
        >
          Nombre:
        </label>
        <input
          className="form-control"
          name="nombre"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <br />
      </div>
      <div
        className="form-group col-md-2 "
        id="grad"
      >
        <label
          htmlFor="grad"
        >
          Graduación:
        </label>
        <input
          className="form-control"
          name="grad"
          onChange={[Function]}
          required={true}
          size="3"
          type="text"
          value=""
        />
        <br />
      </div>
    </div>
    <div
      className="form-group"
      id="sel"
    >
      <label
        htmlFor="sele"
      >
        Selección
      </label>
      <select
        id="tipo"
        name="tipo"
        onChange={[Function]}
        required={true}
      >
        <option
          value="0"
        >
          Seleccione Un tipo
        </option>
        <option
          value="1"
        >
          Ale
        </option>
        <option
          value="2"
        >
          De trigo
        </option>
        <option
          value="3"
        >
          IPA
        </option>
        <option
          value="4"
        >
          Lager
        </option>
        <option
          value="5"
        >
          Lambic
        </option>
        <option
          value="6"
        >
          Pilsen
        </option>
        <option
          value="7"
        >
          Porter
        </option>
        <option
          value="8"
        >
          Stout
        </option>
      </select>
      <br />
    </div>
    <div
      className="form-group "
      id="ingr"
    >
      <label
        htmlFor="ingr"
      >
        Ingredientes:
      </label>
      <br />
      <textarea
        cols="80"
        name="ingr"
        onChange={[Function]}
        rows="4"
        value=""
      />
      <br />
    </div>
    <div
      className="form-group"
      id="check"
    >
      <label>
        <input
          id="chk"
          name="chk"
          onChange={[Function]}
          type="checkbox"
          value="Sin Gluten"
        />
        Sin gluten
      </label>
      <br />
    </div>
    <div
      className="form-group "
      id="pre"
    >
      <label
        htmlFor="precio"
      >
        Precio:
      </label>
      <br />
      <input
        name="precio"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <br />
    </div>
    <div
      className="form-group"
    >
      <button
        align="center"
        className="btn btn-primary"
        disabled={true}
        type="submit"
      >
        Añadir
      </button>
    </div>
  </div>
</form>
`;

exports[`components > Home.page.jsx Comprobamos que se renderiza bien nuestro componente Home 1`] = `
<div
  className="cuerpo"
/>
`;

exports[`components > Listado.jsx Comprobamos que se renderiza bien nuestro componente Listado 1`] = `
<div
  className="listadoContainer"
>
  <div>
    <div
      className="tablaListado titulosTabla"
    >
      <div
        className="elementoLista"
      >
        ID
      </div>
      <div
        className="elementoLista"
      >
        Imagen
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porNombre/"
        >
          Nombre
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porGraduacion/"
        >
          Grad
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porTipo/"
        >
          Tipo
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        Ingredientes
      </div>
      <div
        className="elementoLista"
      >
        Sin Gluten
      </div>
      <div
        className="elementoLista"
      >
        Precio
      </div>
    </div>
  </div>
</div>
`;

exports[`components > PorGraduacion Comprobamos que se renderiza correctamente nuestro componente PorGraduacion 1`] = `
<div
  className="listadoContainer"
>
  <div>
    <div
      className="tablaListado titulosTabla"
    >
      <div
        className="elementoLista"
      >
        ID
      </div>
      <div
        className="elementoLista"
      >
        Imagen
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porNombre"
        >
          Nombre
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porGraduacion"
        >
          Grad
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porTipo/"
        >
          Tipo
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        Ingredientes
      </div>
      <div
        className="elementoLista"
      >
        Sin Gluten
      </div>
      <div
        className="elementoLista"
      >
        Precio
      </div>
    </div>
  </div>
</div>
`;

exports[`components > PorNombre.jsx Comprobamos que se renderiza correctamente nuestro componente PorNombre 1`] = `
<div
  className="listadoContainer"
>
  <div>
    <div
      className="tablaListado titulosTabla"
    >
      <div
        className="elementoLista"
      >
        ID
      </div>
      <div
        className="elementoLista"
      >
        Imagen
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porNombre/"
        >
          Nombre
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porGraduacion/"
        >
          Grad
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porTipo/"
        >
          Tipo
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        Ingredientes
      </div>
      <div
        className="elementoLista"
      >
        Sin Gluten
      </div>
      <div
        className="elementoLista"
      >
        Precio
      </div>
    </div>
  </div>
</div>
`;

exports[`components > PorTipo Comprobamos que se renderiza correctamente nuestro componente PorTipo 1`] = `
<div
  className="listadoContainer"
>
  <div>
    <div
      className="tablaListado titulosTabla"
    >
      <div
        className="elementoLista"
      >
        ID
      </div>
      <div
        className="elementoLista"
      >
        Imagen
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porNombre/"
        >
          Nombre
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porGraduacion/"
        >
          Grad
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        <Link
          replace={false}
          to="/listado/porTipo/"
        >
          Tipo
        </Link>
      </div>
      <div
        className="elementoLista"
      >
        Ingredientes
      </div>
      <div
        className="elementoLista"
      >
        Sin Gluten
      </div>
      <div
        className="elementoLista"
      >
        Precio
      </div>
    </div>
  </div>
</div>
`;
